src/dmd/expressionsem.d

```diff
@@ -11748,3 +11764,28 @@ VarDeclaration makeThis2Argument(const ref Loc loc, Scope* sc, FuncDeclaration f
     vthis2.nestedrefs.push(fd);
     return vthis2;
 }
+
+/*******************************
+ * Make sure that the runtime hook exists.
+ * Params:
+ *      loc = location to use for error messages
+ *      sc = current scope
+ *      id = the hook identifier
+ *      description = what the hook does
+ *      module_ = what module the hook is located in
+ * Returns:
+ *      If the hook was found.
+ */
+bool verifyHookExist(const ref Loc loc, ref Scope sc, Identifier id, string description, Identifier module_ = Id.object)
+{
+    Dsymbol rootSymbol = sc.search(loc, Id.empty, null);
+    Dsymbol moduleSymbol = rootSymbol.search(loc, module_);
```

Comment:
* This could be moved into the if-statement
* `auto` could be used

Reply:
Better like this?
```d
auto rootSymbol = sc.search(loc, Id.empty, null);
if (auto moduleSymbol = rootSymbol.search(loc, module_))
    if (moduleSymbol.search(loc, id))
        return true;
```

Reply:
Possibly, I'm not sure.